client/src/components/UserSearch.tsx

```diff
@@ -49,7 +49,7 @@ export function UserSearch(props: UserProps) {
       defaultActiveFirstOption={false}
       showArrow={defaultValues ? Boolean(defaultValues.length) : false}
       filterOption={false}
-      onSearch={handleSearch}
+      onSearch={debounce(handleSearch, 300)}
```

Comment:
I think it's better to use useDebounce
https://github.com/streamich/react-use/blob/master/docs/useDebounce.md

Reply:
> I think it's better to use useDebounce https://github.com/streamich/react-use/blob/master/docs/useDebounce.md

Like this?
```ts
  const [data, setData] = useState<Person[]>([]);
  const [value, setValue] = useState<string>('');
  const [debouncedValue, setDebouncedValue] = useState<string>('');
  const {
    searchFn = defaultSearch,
    defaultValues,
    keyField,
    showMentor,
    onlyStudentsWithoutMentorShown,
    ...otherProps
  } = props;

  useEffect(() => {
    setData(defaultValues ?? []);
  }, [props.defaultValues]);

  useEffect(() => {
    if (debouncedValue) {
      searchFn(debouncedValue, onlyStudentsWithoutMentorShown).then(data => setData(data));
    } else {
      setData(props.defaultValues ?? []);
    }
  }, [debouncedValue]);

  useDebounce(
    () => {
      setDebouncedValue(value);
    },
    300,
    [value],
  );

  const handleSearch = (value: string) => setValue(value.trim());
```

Reply:
> yes

I have pushed updated solution

Reply:
@ThorsAngerVaNeT it could be done much simpler using `lodash/debounce`

```typescript
import debounce from 'lodash/debounce';
// ...

const handleSearch = debounce(async (value: string) => {
  value = value.trim();
 ...
}, 1000);

Reply:
> @ThorsAngerVaNeT it could be done much simpler using `lodash/debounce`
> 
> ```ts
> import debounce from 'lodash/debounce';
> // ...
> 
> const handleSearch = debounce(async (value: string) => {
>   value = value.trim();
>  ...
> }, 1000);
> ```

You are right. I did it the same way in the beginning, but I used debounce in setting the onSearch prop at line 52. I like the way you did it.

Reply:
You can ignore my comment

Reply:
> You can ignore my comment

Got it :)